app/models/book.rb

```diff
@@ -2,14 +2,20 @@ class Book < ApplicationRecord
   has_many :checkouts
   has_many :users, through: :checkouts
 
-  validates :title, presence: true, uniqueness: { case_sensitive: false}
+  validates :title, presence: true, uniqueness: { case_sensitive: false }
   validates :author, presence: true
 
-  def self.available
-    includes(:checkouts).where(checkouts: { id: nil })
+  def self.unavailable
+    Book.joins(:checkouts).where(checkouts: {closed_at: nil })
   end
 
-  def self.unavailable
-    includes(:checkouts).where.not(checkouts: { id: nil })
+  def self.available
+    Book.joins(
```

Comment:
I think this is fine for this PR. I wouldn't want to chase it much further here given the size of this change already. I have some ideas on how you could refactor this to use active record associations and methods exclusively, but nopthing I'm really positive would work. Maybe we can pair on this later after this change is in.